Tests/SafeDIMacrosTests/InstantiableMacroTests.swift

```diff
@@ -1453,6 +2113,534 @@ import Testing
 			}
 		}
 
+		@Test
+		func declaration_fixit_updatesInitWhenExistingInitIsMissingAccessModifier() {
+			assertMacro {
+				"""
+				@Instantiable
+				public final class UserService: Instantiable {
+					init(a: A, b: B, c: C) {
+						self.a = a
+						self.b = b
+						self.c = c
+					}
+
+					@Received let a: A
+					@Instantiated let b: B
+					@Forwarded let c: C
+				}
+				"""
+			} diagnostics: {
+				"""
+				@Instantiable
+				public final class UserService: Instantiable {
+					init(a: A, b: B, c: C) {
+				 ╰─ 🛑 @Instantiable-decorated type must have a `public` or `open` initializer.
+				    ✏️ Add `public` modifier
+						self.a = a
+						self.b = b
+						self.c = c
+					}
+
+					@Received let a: A
+					@Instantiated let b: B
+					@Forwarded let c: C
+				}
+				"""
+			} fixes: {
+				"""
+				@Instantiable
+				public final class UserService: Instantiable {
+					public init(a: A, b: B, c: C) {
+						self.a = a
+						self.b = b
+						self.c = c
+					}
+
+					@Received let a: A
+					@Instantiated let b: B
+					@Forwarded let c: C
+				}
+				"""
+			} expansion: {
+				"""
+				public final class UserService: Instantiable {
+					public init(a: A, b: B, c: C) {
+						self.a = a
+						self.b = b
+						self.c = c
+					}
+
+					let a: A
+					let b: B
+					let c: C
+
+					public typealias ForwardedProperties = C
+				}
+				"""
+			}
+		}
+
+		@Test
+		func declaration_fixit_updatesInitWhenExistingInitIsMissingAccessModifierWithOtherModifier() {
+			assertMacro {
+				"""
+				@Instantiable
+				public final class UserService: Instantiable {
+					final init(a: A, b: B, c: C) {
+						self.a = a
+						self.b = b
+						self.c = c
+					}
+
+					@Received let a: A
+					@Instantiated let b: B
+					@Forwarded let c: C
+				}
+				"""
+			} diagnostics: {
+				"""
+				@Instantiable
+				public final class UserService: Instantiable {
+					final init(a: A, b: B, c: C) {
+				 ╰─ 🛑 @Instantiable-decorated type must have a `public` or `open` initializer.
+				    ✏️ Add `public` modifier
+						self.a = a
+						self.b = b
+						self.c = c
+					}
+
+					@Received let a: A
+					@Instantiated let b: B
+					@Forwarded let c: C
+				}
+				"""
+			} fixes: {
+				"""
+				@Instantiable
+				public final class UserService: Instantiable {
+					final public init(a: A, b: B, c: C) {
```

Comment:
nit: `public final`? linter will catch it too, so doesn't really matter

Reply:
(and see the line above)

Reply:
🤔 I can prepend instead of postpend since that's likely what folk will want. will play.

Reply:
addressed in 373e4b3dc0ec632626060d6ee9b035278604c458